public/app/themes/justice/inc/core.php

```diff
@@ -59,4 +62,42 @@ public function removeDefaultDashboardWidgets(): void
         remove_meta_box('dashboard_quick_press', 'dashboard', 'side');      // Quick Press/Quick Draft
         remove_meta_box('dashboard_activity', 'dashboard', 'normal');       // Activity
     }
+
+    /**
+     * Handle loopback requests.
+     *
+     * Handle requests to the application host, by sending them to the loopback url.
+     *
+     * @param false|array|WP_Error $preempt
+     * @param array $args
+     * @param string $url
+     * @return false|array|WP_Error
+     */
+
+    public function handleLoopbackRequests(false|array|WP_Error $preempt, array $args, string $url): false|array|WP_Error
+    {
+        $loopback_url = Config::get('WP_LOOPBACK');
+
+        // Do we have a loopback url?
+        if (empty($loopback_url)) {
+            return $preempt;
+        }
+
+        // Is the request url to the application host?
+        if (parse_url($url, PHP_URL_HOST) !== parse_url(get_home_url(), PHP_URL_HOST)) {
+            return $preempt;
+        }
+
+        // Replace the URL.
+        $new_url = str_replace(get_home_url(), $loopback_url, $url);
```

Comment:
Just a thought...
This will fire on every HTTP request called via `WP_Http::request()` 
Before modifying the URL, are we certain that every other type of request has been satisfied? For example; a call to a valid external site?

Reply:
Hey, good point!

This condition will catch if the request url host matches the host of WP_HOME

`if (parse_url($url, PHP_URL_HOST) !== parse_url(get_home_url(), PHP_URL_HOST))`

So, we're 100% only catching requests that are meant for the application.